lib/adiwg/mdtranslator/readers/iso19115_3/modules/module_geographic_extent.rb

```diff
@@ -52,17 +55,18 @@ def self.process_geo_extent(xGeoElem, hGeoExt, hResponseObj)
                   # :geographicElements hash (there's no internal object for these elements...)
                   # :nativeGeoJson array
                   # :computedBbox bash
+
+                  hGeoExt
                end
 
                def self.unpack(xExtent, hResponseObj)
-                  intMetadataClass = InternalMetadata.new
-                  hGeoExt = intMetadataClass.newGeographicExtent
+                  geoExtents = []
 
                   xGeoElems = xExtent.xpath(@@geoElemXPath)
                   xGeoElems.each do |xgeoelem|
-                     process_geo_extent(xgeoelem, hGeoExt, hResponseObj)
+                     geoExtents << process_geo_extent(xgeoelem, hResponseObj)
                   end
-                  hGeoExt
+                  geoExtents
```

Comment:
What purpose does this serve at the end here? is this somehow returning this variable?

Reply:
yes, `geoExtents` will be the function return. ruby doesn't require the "return" statement. I did find a situation where explicitly calling "return" produced the correct result but that was probably coincidental